Sources/Floating Point Conversion.swift

```diff
@@ -22,23 +26,69 @@ extension BigUInt {
     public init<T: BinaryFloatingPoint>(_ source: T) {
         self.init(exactly: source.rounded(.towardZero))!
     }
+
+    #if canImport(Foundation)
+    public init?(exactly source: Decimal) {
+        guard source.isFinite else { return nil }
+        guard !source.isZero else { self = 0; return }
+        guard source.sign == .plus else { return nil }
+        assert(source.floatingPointClass == .positiveNormal)
+        guard source.exponent >= 0 else { return nil }
+        let intMaxD = Decimal(UInt.max)
+        let intMaxB = BigUInt(UInt.max)
+        var start = BigUInt()
+        var value = source
+        while value >= intMaxD {
+            start += intMaxB
+            value -= intMaxD
+        }
+        start += BigUInt((value as NSNumber).uintValue)
+        self = start
+    }
```

Comment:
This is not always exactly right? It is missing some kind of check.
Maybe we could check for the exponent to be greater than or equal to zero.

Reply:
Oh good - small things are easy :)

Do you mean line 36? Or is there something else I'm missing? I'm pretty sure this *should* be exact, though. (Tho I'm not well versed in the intricacies of floating point, etc)

Reply:
I didnt see that line my bad.